scripts/examples/hosting/update-single-file/src/index.ts

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env node
+/**
+ * Copyright (c) 2022 Google LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import crypto from "node:crypto";
+import fs from "node:fs";
+import path from "node:path";
+import zlib from "node:zlib";
+
+import debugPkg from "debug";
+import minimist from "minimist";
+import { GoogleAuth } from "google-auth-library";
+
+const debug = debugPkg("update-single-file");
+
+const HOSTING_URL = "https://firebasehosting.googleapis.com/v1beta1";
+
+async function main(): Promise<void> {
+  const argv = minimist<{ project?: string; site?: string }>(process.argv.slice(2));
+  const PROJECT_ID = argv.project;
+  if (!PROJECT_ID) {
+    throw new Error(`--project must be provided.`);
+  }
+  const SITE_ID = argv.site || PROJECT_ID;
+  const files = Array.from(argv._);
+  console.log(`Deploying files:`);
+  for (const f of files) {
+    console.log(`- ${f}`);
+  }
+
+  const filesByHash: Record<string, string> = {};
+  for (const file of files) {
+    const hasher = crypto.createHash("sha256");
```

Comment:
Maybe abstract the "gzip and hash a single file" part into a function? Would be helpful to have it separated for exemplar reasons I think.

Reply:
Done.